src/cache_interface.jl

```diff
@@ -1,15 +1,7 @@
-function get_p(::AbstractOptimizationCache)
-    error("`get_p`: method has not been implemented for the cache")
-end
-function get_observed(::AbstractOptimizationCache)
-    error("`get_observed`: method has not been implemented for the cache")
-end
-function get_syms(::AbstractOptimizationCache)
-    error("`get_syms`: method has not been implemented for the cache")
-end
-function get_paramsyms(::AbstractOptimizationCache)
-    error("`get_paramsyms`: method has not been implemented for the cache")
-end
+get_p(cache::AbstractOptimizationCache) = cache.p
+get_observed(cache::AbstractOptimizationCache) = cache.f.observed
+get_syms(cache::AbstractOptimizationCache) = cache.f.syms
+get_paramsyms(cache::AbstractOptimizationCache)= cache.f.paramsyms
```

Comment:
these aren't part of the interface. If they are to be added to the interface, it needs to be done correctly.

Reply:
Sry! How should I do it then?